CMakeLists.txt

```diff
@@ -87,7 +87,7 @@ set(GENERATED_DIRECTORY ${PROJECT_BINARY_DIR}/generated)
 file(MAKE_DIRECTORY ${GENERATED_DIRECTORY}/include)
 
```

Comment:
@marc-hb I wouldn't change the name here. kconfig.h can be later used for kconfig related stuff.

You can use this PR only to verify that there are no config.h left but later drop it.

Reply:
This _is_ kconfig stuff? I can pick any other name. Changing it is important to make sure no one keeps including it.

Reply:
This is generated kconfig stuff. I was thinking that sometime in the future we might add kconfig per se stuff like this:
https://elixir.bootlin.com/linux/latest/source/include/linux/kconfig.h

What about renaming it autoconf.h ?

Reply:
I went for autoconfig.h because autoconf is already something different and including *config.h in the name increases the chances of it being "grepped". Let me know what you think.

Reply:
Ok, it is an acceptable solution. Looks good to me.